guppylang/diagnostic.py

```diff
@@ -132,7 +132,11 @@ def rendered_title(self) -> str:
 
     def add_sub_diagnostic(self, sub: "SubDiagnostic") -> Self:
         """Adds a new sub-diagnostic."""
-        if self.span and sub.span and to_span(sub.span).file != to_span(self.span).file:
+        if (
+            self.span is not None
+            and sub.span is not None
+            and to_span(sub.span).file != to_span(self.span).file
+        ):
```

Comment:
Drive-by fix that was discovered by one of the tests:

The truthy check before tried to invoke `__len__` which fails for multi-line spans